### PR TITLE
prometheus exporter: export application level metrics

### PR DESCRIPTION
**Pull request by skaes**
This PR adds application level metrics to the prometheus exporter

### CODE REVIEW

File: go/prometheusexporter/collector/collector.go [hunk 1]
```diff
@@ -284,11 +376,21 @@ func (c *Collector) registerJobExecutionHistogramVec(stream *util.Stream) {
 		// instance always set to the empty string
 		[]string{"app", "env", "action", "instance"},
 	)
-	c.registry.MustRegister(c.jobExecutionHistogramVec)
+	c.registry.MustRegister(c.actionMetrics.jobExecutionHistogramVec)
+	c.applicationMetrics.jobExecutionHistogramVec = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name:    "logjam:application:job_execution_time_distribution_seconds",
+			Help:    "logjam background job execution time distribution by action",
```
Comment (0robustus1) [inline go/prometheusexporter/collector/collector.go:383]:
```suggestion
			Help:    "logjam background job execution time distribution by application",
```

File: go/prometheusexporter/collector/collector.go [hunk 2]
```diff
@@ -310,98 +422,125 @@ func (c *Collector) registerAjaxHistogramVec(stream *util.Stream) {
...
+		},
+		// instance always set to the empty string
+		[]string{"app", "env", "instance"},
+	)
+	c.registry.MustRegister(c.applicationMetrics.ajaxHistogramVec)
 }
 
-func (c *Collector) registerActionCallsTotalVec() {
-	c.actionCallsTotalVec = prometheus.NewCounterVec(
+func (c *Collector) registerActionsProcessedTotalVec() {
```
Comment (0robustus1) [inline go/prometheusexporter/collector/collector.go:438]: We could rename the function to `registerTransactionsTotalVec` to keep it in line with the other ones.

File: go/prometheusexporter/collector/collector.go [hunk 3]
```diff
@@ -473,33 +612,63 @@ func (c *Collector) deleteLabels(name string, labels prometheus.Labels) bool {
...
-		deleted = c.pageHistogramVec.Delete(labels)
+		deleted = c.actionMetrics.pageHistogramVec.Delete(labels)
 	case "logjam:action:ajax_time_distribution_seconds":
-		deleted = c.ajaxHistogramVec.Delete(labels)
-	case "logjam:action:action_calls_total":
-		deleted = c.actionCallsTotalVec.Delete(labels)
+		deleted = c.actionMetrics.ajaxHistogramVec.Delete(labels)
+	case "logjam:action:transactions_total":
+		deleted = c.actionMetrics.transactionsTotalVec.Delete(labels)
+	case "logjam:application:http_response_time_summary_seconds":
```
Comment (0robustus1) [inline go/prometheusexporter/collector/collector.go:632]: I don't think adding the `logjam:application:` metrics here will have any effect.
`labels` always contains `"action"`, which means that `Delete(labels)` on an application metric will never match a time-series and effectively be a no-op.

File: go/prometheusexporter/collector/collector.go (hunk 1 above)
Comment (skaes) [reply go/prometheusexporter/collector/collector.go:383]: Fixed

File: go/prometheusexporter/collector/collector.go (hunk 2 above)
Comment (skaes) [reply go/prometheusexporter/collector/collector.go:438]: fixed

File: go/prometheusexporter/collector/collector.go (hunk 3 above)
Comment (skaes) [reply go/prometheusexporter/collector/collector.go]: you're right, this code is superfluous. I got rid of it.

**Pull request merged**